src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs

```diff
@@ -649,9 +653,92 @@ void vm_CurrentOffsetChanged(object sender, EventArgs e)
 
         void vm_ZoomChanged(object sender, EventArgs e)
         {
-            zoomBorder.SetZoom((e as ZoomEventArgs).Zoom);
-            if (PortContextMenu.IsOpen) DestroyPortContextMenu();
+            var newZoomScale = (e as ZoomEventArgs).Zoom;
+            zoomBorder.SetZoom(newZoomScale);
+            if (PortContextMenu.IsOpen)
+            {
+                DestroyPortContextMenu();
+            }
+
+            CheckZoomScaleAndApplyNodeViewCache(newZoomScale);
+        }
+
+        #region NodeView_BitmapCache
+        private void CheckZoomScaleAndApplyNodeViewCache(double newZoomScale)
+        {
+            //disable bitmap caching if max zoom scale set to 0, or feature flag was unable to fetch;
+            if (ViewModel.MaxZoomScaleForBitmapCache == 0) return;
+
+            if (!ViewModel.StopNodeViewOpacityAnimations)
+            {
+                if (currentRenderScale > 0) // number of nodes reduced below max threshold
+                {
+                    Dispatcher.BeginInvoke(ClearNodeViewCache, DispatcherPriority.Normal);
+                }
+
+                return;
+            }
+
+            if (newZoomScale > ViewModel.MaxZoomScaleForBitmapCache)
+            {
+                if (currentRenderScale > 0)
+                {
+                    Dispatcher.BeginInvoke(ClearNodeViewCache, DispatcherPriority.Normal);
+                }
+                
+                return;
+            }
+
+            var newRenderScale = newZoomScale <= .2 ? .2 : newZoomScale <= .5 ? .5 : 1;
+
+            if (Math.Abs(newRenderScale - currentRenderScale) <= 0.01)
+            {
+                return;
+            }
+
+            currentRenderScale = newRenderScale;
+            Dispatcher.BeginInvoke(UpdateNodeViewCacheScale, DispatcherPriority.Normal);
+        }
+
+        /// <summary>
+        /// Clear the node view bitmap cache
+        /// </summary>
+
+        private void ClearNodeViewCache()
+        {
+            var nodes = this.ChildrenOfType<NodeView>();
```

Comment:
hmm - have you tested this change on a custom node workspace?

Reply:
yes

![DynamoSandbox_nEllgn1Ed9](https://github.com/user-attachments/assets/c57d82f7-7f24-42e0-8d20-57077272b1af)